internal/service/certificatesigningrequest.go

```diff
@@ -231,15 +290,24 @@ func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, r
 		Message: "Approved",
 	}
 	err = h.store.CertificateSigningRequest().UpdateConditions(ctx, orgId, request.Name, []api.Condition{approvedCondition})
+	if err != nil {
+		if errors.Is(err, flterrors.ErrResourceNotFound) {
+			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
+		} else {
```

Comment:
optional: since you return no need for else?

```
           if err != nil {
		if errors.Is(err, flterrors.ErrResourceNotFound) {
			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
		} 
		return nil, err
           }